reference_designs/ipu-xrt/matrix_multiplication/test.cpp

```diff
@@ -0,0 +1,241 @@
+//===- test.cpp -------------------------------------------000---*- C++ -*-===//
+//
+// This file is licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// Copyright (C) 2023, Advanced Micro Devices, Inc.
+//
+//===----------------------------------------------------------------------===//
+
+#include <boost/program_options.hpp>
+#include <cstdint>
+#include <cstdlib>
+#include <ctime>
+#include <fstream>
+#include <iomanip>
+#include <iostream>
+#include <sstream>
+#include <string>
+#include <vector>
+
+#include "xrt/xrt_bo.h"
+#include "xrt/xrt_device.h"
+#include "xrt/xrt_kernel.h"
+
+#define M 128
+#define K 128
+#define N 128
+
+#define A_VOLUME M *K
+#define B_VOLUME N *K
+#define C_VOLUME M *N
+
+#define A_DATATYPE int16_t
+#define B_DATATYPE int16_t
+#define C_DATATYPE int16_t
```

Comment:
Say it in C++
```suggestion
using A_DATATYPE = int16_t;
using B_DATATYPE = int16_t;
using C_DATATYPE = int16_t;
```
or even less C:
```suggestion
using A_DATATYPE = std::int16_t;
using B_DATATYPE = std::int16_t;
using C_DATATYPE = std::int16_t;
```